### DISCUSSION TITLE
astro with astro-i18next on vercel/serverless yields error on build

### DISCUSSION THREAD

**Discussion by fprl** [vercel/vercel#8992] [Help] [answered]
Hi there, I'm trying to build my Astro site but I can't deploy it to `vercel/serverless` since version `1.0.0-beta.13`. I know this is related to that package but the error is related to esbuild and vercel deployment, so maybe you can help me with this:

```
14:40:25 [build] Rearranging server assets...
[@astrojs/vercel] The module "pnpapi" inside the file "/Users/me/me/Work/project/node_modules/esbuild/lib/main.js" couldn't be resolved. This may not be a problem, but it's worth checking.
 error   Failed to parse /Users/me/me/Work/project/node_modules/esbuild-darwin-arm64/bin/esbuild as script:
  Unexpected character '�' (1:0)
```

Do you know what could be happening?

**Comment by amyegan** [comment]
Thanks for reaching out @francoromanol! Can you tell me more about your project setup? 

I don't have enough information to know an exact cause, but these are a few quick ideas:
- There is an esbuild issue with a similar problem https://github.com/evanw/esbuild/issues/2463
- Stack Overflow post "[Parsing error: Unexpected character '#' while declaring priavate member in JavaScript class](https://stackoverflow.com/questions/72297997/parsing-error-unexpected-character-while-declaring-priavate-member-in-javas)" may offer a solution to the parsing problem
- If you're using Yarn, it may be important to know that the Vercel build pipeline fully supports [Yarn 1](https://classic.yarnpkg.com/lang/en/) for all deployment types, but [Yarn 2](https://yarnpkg.com/) is only supported for static builds. [Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions) and SSR (Server-side Rendering) do not work with Yarn 2.

If you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), I am happy to take a closer look!

**Reply by fprl** [reply]
Hi @amyegan, thank you so much for your help.

I would say Yarn it's not the problem in this case because its giving the same error with npm (reproducible example) but I will check the first two issues to see if they are related to what's happening with astro-i18next package (since `1.0.0-beta.13`).

Here's the [minimal reproducible example](https://github.com/francoromanol/astro-i18next-vercel), and I attached an image of the deployment log in Vercel.

<img width="508" alt="image" src="https://user-images.githubusercontent.com/68303277/205016228-48daf98d-4248-4e2e-bd76-d73d19241efa.png">

**PS:** This is not happening when building with vercel/static.

**Reply by amyegan** [reply]
Thanks for sharing this example! I suspect this error is related to issue https://github.com/vercel/vercel/issues/7908. There is a change being made right now to address TypeScript ESM support (draft PR https://github.com/vercel/vercel/pull/7954) which should be available very soon

**Reply by fprl** [reply]
Nice, thanks :)
It's weird that is not working since two versions, I don't know what could be pushed that broke this.

**Comment by fprl** [comment]
Hey @amyegan, can I ask you other thing about deploying Astro on vercel/edge adapter? I'm getting a blank page.

Attached logs from function
```
[GET] /
TypeError: Failed to execute 'encode' on 'TextEncoder': parameter 1 is not of type 'String'.
    at worker.js:113:2401
    at worker.js:136:12

[GET] /
TypeError: Failed to execute 'encode' on 'TextEncoder': parameter 1 is not of type 'String'.
```

It's weird because on dev it works, but the [deployed version](https://covered-astro-3728oy5fb-coveredibiza.vercel.app/) is not returning anything.

**Reply by amyegan** [reply]
It looks like you have a type error in your edge function. You'll need to check that you're getting the expected data in that function.

**Reply by fprl** [reply]
Hi @amyegan, I tried removing things and I realized this error is coming from typescript paths defined in ts.config as once I import a component using this syntax it starts failing.

```
{
  "extends": "astro/tsconfigs/base",
  "compilerOptions": {
    "allowJs": true,
    "jsx": "preserve",
    "baseUrl": "src",
    "paths": {
      "@src/*": ["*"],
      "@layouts/*": ["layouts/*"],
      "@components/*": ["components/*"],
      "@features/*": ["features/*"],
      "@lib/*": ["lib/*"],
      "@locales/*": ["locales/*"],
      "@composables/*": ["composables/*"],
      "@utils/*": ["utils/*"]
    },
    "types": ["unplugin-icons/types/vue"]
  }
}
```

`import Hero from '@components/statics/Hero.astro';`

Any tip here?

**Reply by amyegan** [reply]
It still could also be related to issue https://github.com/vercel/vercel/issues/7908, since that involves TypeScript ESM support. Or it could be a normal TypeScript error in the adapter. I didn't see `TextEncoder` in the MRE, but if you're using it in your real site code then an incompatible parameter type could be the source of the error.

I think this question is worth asking the [Astro community on Discord](https://astro.build/chat) as well since you're using `astro/tsconfigs/base`. They may recognize the error and have a solution.

**Reply by fprl** [reply]
I will check it and also ask in Astro discord.

Thanks a have a nice sunday :)

**Comment by amyegan** [comment, marked as answer]
[Support for ESM with TS](https://github.com/vercel/vercel/pull/7954) was released! You may need to make some of the `tsconfig` and CJS/ESM adjustments outlined in https://github.com/vercel/community/discussions/1225#discussioncomment-4642381. If you see any more problems with this, please let me know 🙏
